src/parser.rs

```diff
@@ -3136,6 +3136,9 @@ impl<'a> Parser<'a> {
                     name,
                     cascade,
                 }
+            } else if self.parse_keyword(Keyword::PRIMARY) {
```

Comment:
@ding-young `ALTER TABLE tb DROP PRIMARY` isn't allowed and, therefore, shouldn't be acceptable. 

The `let _` in the next line is permissive over this.

Also, I think this is a MySQL-specific statement. Could you please add the validation to accept only `MySQL` and `GenericDialect`, please?